liberty-maven-plugin/src/it/dev-it/src/test/java/net/wasdev/wlp/test/dev/it/BaseDevTest.java

```diff
@@ -0,0 +1,159 @@
+      assertTrue(tempProj.exists());
+      
+      basicDevProj = new File("../resources/basic-dev-project");
+      assertTrue(basicDevProj.exists());
+
+      FileUtils.copyDirectoryStructure(basicDevProj, tempProj);
+      assertTrue(tempProj.listFiles().length > 0);
+   }
+
+   @After
+   public void cleanUp() throws Exception {
+      if (tempProj != null && tempProj.exists()) {
+         FileUtils.deleteDirectory(tempProj);
+      }
+   }
+
+   @Test
+   public void basicTest() throws Exception {
+
+      // run dev mode on project
+      ProcessBuilder builder = new ProcessBuilder();
+      builder.directory(tempProj);
+      String processCommand = "mvn liberty:dev";
+
+      String os = System.getProperty("os.name");
+      if (os != null && os.toLowerCase().startsWith("windows")) {
+         builder.command("CMD", "/C", processCommand);
+      } else {
+         builder.command("bash", "-c", processCommand);
+      }
+
+      File logFile = new File(basicDevProj, "/logFile.txt");
+      Files.write(logFile.toPath(), "".getBytes());
+
+      builder.redirectOutput(logFile);
+      builder.redirectError(logFile);
+      Process process = builder.start();
+      assertTrue(process.isAlive());
+
+      OutputStream stdin = process.getOutputStream();
+
+      BufferedWriter writer = new BufferedWriter(new OutputStreamWriter(stdin));
+      Thread.sleep(40000); // wait for dev mode to start up
+
+      // check that the server has started
+      int startTimeout = 100000;
+      int startWaited = 0;
+      boolean startFlag = false;
+      while (!startFlag && startWaited <= startTimeout) {
+         int sleep = 10;
+         Thread.sleep(sleep);
+         startWaited += sleep;
+         if (readFile("CWWKF0011I", logFile) == true) {
+            startFlag = true;
+         }
+     }
+   
+      // verify that the target directory was created
+      File targetDir = new File(tempProj, "/target");
+      assertTrue(targetDir.exists());
+
+      // modify a java file
+      File srcHelloWorld = new File(tempProj, "/src/main/java/com/demo/HelloWorld.java");
+      File targetHelloWorld = new File(targetDir, "/classes/com/demo/HelloWorld.class");
+      assertTrue(srcHelloWorld.exists());
+      assertTrue(targetHelloWorld.exists());
+
+      long lastModified = targetHelloWorld.lastModified();
+      String str = "// testing";
+      BufferedWriter javaWriter = new BufferedWriter(new FileWriter(srcHelloWorld, true));
+      javaWriter.append(' ');
+      javaWriter.append(str);
+
+      javaWriter.close();
+
+      Thread.sleep(2000); // wait for compilation
+      boolean wasModified = targetHelloWorld.lastModified() > lastModified;
+      assertTrue(wasModified);
+
+      // shut down dev mode
+      writer.write("exit"); // trigger dev mode to shut down
+      writer.flush();
+      writer.close();
+      Thread.sleep(2000); // wait for dev mode to shut down
+
+      process.waitFor();
```

Comment:
Use with a timeout in case the process doesn't actually end.  See `Process.waitFor(long timeout, TimeUnit unit)`